File: leetCodePractice/tree/inorderTreeTraversal.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def inorderTraversal(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        self.result = []
        self.dfs(root)
        return self.result
        
    def dfs(self, root):
        if root is None: 
            return
        if root.left:
            self.dfs(root.left)
        if not root.val is None:
            self.result.append(root.val)
        if root.right:
            self.dfs(root.right)
        return

File: leetCodePractice/tree/inorderTreeTraversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
 var inorderTraversal = function(root) {
    let result = [];
    
    const dfs = (root) => {
        if (root === null) {
            return;
        }
        console.log(root.val)
        if (root.left) {
            dfs(root.left);
        }
        if (root.val !== null) {
            result.push(root.val);
        }
        if (root.right) {
            dfs(root.right);
        }
        return;
    }
    dfs(root);
    return result;
    
};